src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example2.java

```diff
@@ -0,0 +1,25 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="PatternVariableName">
+        <property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/>
+    </module>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.patternvariablename;
+
+// xdoc section -- start
+class Example2 {
+  Example2(Object o1){
+    if (o1 instanceof String STR) { // violation
```

Comment:
Please make all inputs to be the same by java code and different only in comments